test/unit/ngsiv2/provisioning/updateProvisionedDevices-test.js

```diff
@@ -0,0 +1,411 @@
+                    done();
+                });
+            });
+        });
+    });
+    describe('When an update request arrives with a new Device ID', function() {
+        var optionsUpdate = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/Light1',
+            method: 'PUT',
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            },
+            json: utils.readExampleFile(
+                './test/unit/examples/deviceProvisioningRequests/updateProvisionDeviceWithId.json')
+        };
+
+        it('should raise a 400 error', function(done) {
+            request(optionsUpdate, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(400);
+                done();
+            });
+        });
+    });
+    describe('When a wrong update request payload arrives', function() {
+        var optionsUpdate = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/Light1',
+            method: 'PUT',
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            },
+            json: utils.readExampleFile(
+                './test/unit/examples/deviceProvisioningRequests/updateProvisionDeviceWrong.json')
+        };
+
+        it('should raise a 400 error', function(done) {
+            request(optionsUpdate, function(error, response, body) {
+                should.not.exist(error);
+                response.statusCode.should.equal(400);
+                done();
+            });
+        });
+    });
+
+    describe('When a device is provisioned without attributes and new ones are added through an update', function() {
+        var optionsUpdate = {
+                url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/MicroLight2',
+                method: 'PUT',
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                },
+                json: utils.readExampleFile('./test/unit/examples/deviceProvisioningRequests/updateMinimumDevice.json')
+            },
+            optionsGetDevice = {
+                url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/MicroLight2',
+                method: 'GET',
+                headers: {
+                    'fiware-service': 'smartGondor',
+                    'fiware-servicepath': '/gardens'
+                }
+            };
+
+        beforeEach(function(done) {
+            nock.cleanAll();
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/entities?options=upsert')
```

Comment:
Mock using post without payload. Comment explaining it needed?

Reply:
Fixed with fbc86fa76f4561b711a63053695efd21941cf4e5